src/integrators/ResNN.jl

```diff
@@ -61,7 +67,7 @@ function  apply(this::ResNN{T},theta::Array{T},Y0::Array{T},doDerivative=true) w
 
     Ydata = zeros(T,0,nex)
     for i=1:this.nt
-        Z,dummy,tmp[i,2] = apply(this.layer,theta[:,i],Y,doDerivative)
+        Z,dummy,tmp[i,2] = apply(this.layer,theta[:,i],Y,doDerivative,tmp[i,2])
```

Comment:
tmp[i, 2] creates a sub array that copies the data out of tmp into a temporary variable. I suspect that this means you would not really be saving many memory allocation passing the input like this. 

I suspect would need to either pass `view(tmp, i, 2)` to create a reference to that section in memory, or pass `tmp, i, 2' as three input arguments and modify tmp in place within the function's scope to accomplish this with no allocations.